integration/e2e_test.go

```diff
@@ -317,28 +363,108 @@ func deployMicrobot(t testing.TB) {
+		"--set", "operator.replicas=1",
+		"--set", "ipam.operator.clusterPoolIPv4PodCIDRList=10.1.0.0/16",
+		"--set", "nodePort.enabled=true",
+	}
+
+	t.Logf("running command: %s", strings.Join(command, " "))
+	cmd := exec.Command(command[0], command[1:]...)
+	cmd.Env = append(cmd.Env, "KUBECONFIG="+KUBECONFIG)
+	outputBytes, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Error(string(outputBytes))
+		t.Fatalf("Failed to deploy cilium from heml chart. %s", err)
+	}
+}
+
+// validateCilium checks a deployment of cilium daemon set.
+func validateCilium(t testing.TB) {
+	t.Log("Validate Cilium")
+
+	//check control plane machine exists
+	attempt := 0
+	maxAttempts := 120
+	machines := 0
+	command := []string{"kubectl", "get", "machine", "--no-headers"}
+	for {
+		t.Logf("running command: %s", strings.Join(command, " "))
+		cmd := exec.Command(command[0], command[1:]...)
+		outputBytes, err := cmd.CombinedOutput()
+		output := string(outputBytes)
+		if err != nil {
+			t.Log(output)
+			if attempt >= maxAttempts {
+				t.Fatal(err)
+			} else {
+				attempt++
+				t.Log("Retrying")
+				time.Sleep(10 * time.Second)
+			}
+		} else {
+			machines = strings.Count(output, "\n")
+			if machines > 0 {
+				break
+			}
+		}
+	}
+
+	t.Log("Checking Cilium daemon set")
+	command = []string{"kubectl", "--kubeconfig=" + KUBECONFIG, "-n", "kube-system", "wait", "ds/cilium", fmt.Sprintf("--for=jsonpath={.status.numberAvailable}=%d", machines)}
+	for {
+		cmd := exec.Command(command[0], command[1:]...)
+		outputBytes, err := cmd.CombinedOutput()
+		if err != nil {
+			t.Logf("running command: %s", strings.Join(command, " "))
+			t.Log(string(outputBytes))
+			if attempt >= maxAttempts {
+				t.Fatal(err)
+			} else {
+				attempt++
+				t.Log("Retrying")
+				time.Sleep(20 * time.Second)
+			}
 		} else {
 			break
 		}
 	}
 }
 
 // upgradeCluster upgrades the cluster to a new version based on the upgrade strategy.
-func upgradeCluster(t testing.TB, upgrade_strategy string) {
+func upgradeCluster(t testing.TB, upgradeStrategy string) {
 
-	version, control_plane_name, control_plane_type, worker_deployment_name,
-		worker_deployment_type := getUpgradeEnvVars(t)
+	version := "v1.28.0"
+	controlPlaneName := "test-ci-cluster-control-plane"
+	controlPlaneType := "microk8scontrolplanes.controlplane.cluster.x-k8s.io"
+	workerDeploymentName := "test-ci-cluster-md-0"
+	workerDeploymentType := "machinedeployments.cluster.x-k8s.io"
 
-	t.Logf("Upgrading cluster to %s via %s", version, upgrade_strategy)
-	// Patch contol plane machine upgrades based on type of upgrade strategy.
-	outputBytes, err := controlPlanePatch(control_plane_name, control_plane_type, version, upgrade_strategy)
+	t.Logf("Upgrading cluster to %s via %s", version, upgradeStrategy)
+	// Patch control plane machine upgrades based on type of upgrade strategy.
+	outputBytes, err := controlPlanePatch(controlPlaneName, controlPlaneType, version, upgradeStrategy)
```

Comment:
the output bytes are not used anywhere, omit it and also inline the error check:
`if _, err := controlPlanePatch(...); err != nil {...` and
`if _, err := workerPatch(...); err != nil {...`